pkg/secret/secret.go

```diff
@@ -0,0 +1,165 @@
+	return SecretsService{
+		client: client,
+		helper: helper,
+	}
+}
+
+// ListNamesAndLabels return the names and labels of all secrets
+func (ss SecretsService) ListNamesAndLabels() (resourceNames, labels []string, err error) {
+	secrets, err := ss.List(metav1.ListOptions{})
+	var objectMetas []metav1.ObjectMeta
+	for _, s := range secrets {
+		objectMetas = append(objectMetas, s.ObjectMeta)
+	}
+	secretNames, labels := util.GetNamesAndLabels(objectMetas)
+	return secretNames, labels, nil
+}
+
+// List returns a list of secrets from a namespace
+func (ss SecretsService) List(listOptions metav1.ListOptions) ([]v1.Secret, error) {
+	secrets, err := ss.client.List(listOptions)
+	if err != nil {
+		return nil, err
+	}
+	return secrets.Items, nil
+}
+
+// GetUnused returns unused resources
+func (ss SecretsService) GetUnused(namespace string, resources []v1.Secret) (unusedResources []v1.Secret, funcErr error) {
+	var usedSecrets []v1.Secret
+	funk.ForEach(openshift.PredefinedResources, func(predefinedResource schema.GroupVersionResource) {
+		funk.ForEach(resources, func(secret v1.Secret) {
+
+			secretName := secret.GetName()
+
+			if funk.Contains(usedSecrets, secret) {
+				// already marked as existing, skip this
+				return
+			}
+			contains, err := ss.helper.ResourceContains(namespace, secretName, predefinedResource)
+			if err != nil {
+				funcErr = err
+				return
+			}
+
+			if contains {
+				usedSecrets = append(usedSecrets, secret)
+			}
+		})
+	})
+
+	for _, resource := range resources {
+		if !funk.Contains(usedSecrets, resource) {
+			unusedResources = append(unusedResources, resource)
+		}
+	}
+
+	return unusedResources, funcErr
+}
+
+// Delete removes secrets
+func (ss SecretsService) Delete(secrets []v1.Secret) {
+	for _, resource := range secrets {
+		namespace := resource.GetNamespace()
+		kind := "Secret"
+		name := resource.GetName()
+
+		log.Infof("Deleting %s %s/%s", kind, namespace, name)
+
+		err := ss.client.Delete(name, &metav1.DeleteOptions{})
+
+		if err != nil {
+			log.WithError(err).Errorf("Failed to delete %s %s/%s", kind, namespace, name)
+		}
+	}
+}
+
+//FilterByTime returns secrets which are older than specified date
+func (ss SecretsService) FilterByTime(secrets []v1.Secret, olderThan time.Time) (filteredResources []v1.Secret) {
+	log.WithFields(log.Fields{
+		"olderThan": olderThan,
+		"resources": secrets,
+	}).Debug("Filtering resources older than the specified time")
+
+	for _, resource := range secrets {
+		lastUpdatedDate := resource.GetCreationTimestamp().Time
+		if lastUpdatedDate.Before(olderThan) {
+			filteredResources = append(filteredResources, resource)
+		} else {
+			name := resource.GetName()
+			log.WithField("name", name).Debugf("Filtered resource %s", name)
+		}
+	}
+
+	return filteredResources
+}
+
+// FilterByMaxCount keep at most n newest resources. The list of secrets is sorted in descending ordered in
```

Comment:
`// FilterByMaxCount returns the latest resources until limited by <keep>. The list of secrets is sorted by CreationTimestamp, with newest entries first.`